superset/assets/src/dashboard/actions/datasources.js

```diff
@@ -46,7 +46,7 @@ export function fetchDatasourceMetadata(key) {
     return SupersetClient.get({
       endpoint: `/superset/fetch_datasource_metadata?datasourceKey=${key}`,
     })
-      .then(data => dispatch(setDatasource(data, key)))
+      .then(({ json }) => dispatch(setDatasource(json, key)))
```

Comment:
can we attach response of fetch to validate that it does have a property named json that need to go here?

Reply:
I didn't get your point ??

SupersetClient have fix format for response parser with these property...

Reply:
What is the response being fetched from endpoint '/superset/....? datasourcKey=$key.

Do we have a network response?

I understand that this is fixed fromat but I don't have sample to see format